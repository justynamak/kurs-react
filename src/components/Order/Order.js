import React, { Component } from "react";
import OrderView from "../Views/OrderView";

class Order extends Component {
  render() {
    const order = this.props.order.map(book => (
      <OrderView
        key={book.name}
        book={book}
        removeFromOrder={this.props.removeFromOrder}
      />
    ));
    return (
      <div className="order col-md-3">
        {order.length ? <ul>{order}</ul> : null}
      </div>
    );
  }
}
export default Order;
