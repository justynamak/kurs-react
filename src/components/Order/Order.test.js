import React from "react";
import ReactDOM from "react-dom";
import Order from "./Order";
import Enzyme, { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Tests rendered", () => {
  it("Order rendered without problem", () => {
    const div = document.createElement("div");
    const order = [];
    ReactDOM.render(<Order order={order} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Snapshot matches", () => {
    const order = [];
    const wrapper = shallow(<Order order={order} />);

    expect(wrapper).toMatchSnapshot();
  });
});
