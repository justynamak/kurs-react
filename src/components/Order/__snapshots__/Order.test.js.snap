// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests rendered Snapshot matches 1`] = `ShallowWrapper {}`;
